src/components/common/YourCompany/EntityName.vue

```diff
@@ -406,9 +443,9 @@ export default class EntityName extends Mixins(NameRequestMixin) {
     // reset name request
     this.setNameRequest({
       legalType: this.getEntitySnapshot.businessInfo.legalType,
-      legalName: this.getEntitySnapshot.businessInfo.legalName,
-      nrNumber: this.getEntitySnapshot.businessInfo.nrNumber
-    })
+      nrNum: this.getEntitySnapshot.businessInfo.nrNumber
+    } as any)
+    this.setNameRequestLegalName(this.getEntitySnapshot.businessInfo.legalName)
```

Comment:
Wonder if we should add optional chaining to the getEntitySnapshot her and above. As in `this.getEntitySnapshot?.businessInfo?.legalName`.

Reply:
The previous code didn't, so I didn't use it here, either.

There is a catch to using optional chaining: if any of the objects is falsy, the expression has a value of undefined. Also, this can hide issues with your objects.

For these reasons, I lean towards _not_ using optional chaining if possible.

Reply:
Got it, makes sense. Thank you!